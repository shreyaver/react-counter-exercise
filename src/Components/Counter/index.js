import React from 'react';
import Button from '../Button';
import './Counter.css';

const Counter = ({ counterValue, handleAddClick, handleSubtractClick }) => {
  return (
    <div className = 'Counter'> 
    <p className = 'Counter-display'>Counter: {counterValue}</p>
    <Button caption = 'Add' onClick = {handleAddClick} />
    <Button caption = 'Subtract' onClick = {handleSubtractClick}/>
    </div>
  )
}
  export default Counter;
