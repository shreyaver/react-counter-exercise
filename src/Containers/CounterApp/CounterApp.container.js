import { incrementActionGenerator, decrementActionGenerator } from '../../redux/index.actions';
import { connect } from 'react-redux';
import Counter from '../../Components/Counter';

export const mapStateToProps = (state) => {
    return { counterValue: state.counterValue };
}
export const mapDispatchToProps = (dispatch) => {
    return { handleAddClick: () => dispatch(incrementActionGenerator()),
        handleSubtractClick: () => dispatch(decrementActionGenerator()),
    }
}
export const CounterApp = connect(mapStateToProps, mapDispatchToProps)(Counter);