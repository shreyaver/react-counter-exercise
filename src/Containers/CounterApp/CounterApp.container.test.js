import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { createMockStore } from 'redux-test-utils';
import { CounterApp } from './CounterApp.container';
import Counter from '../../Components/Counter';
import { incrementActionGenerator, decrementActionGenerator } from '../../redux/index.actions';

describe('CounterApp', () => {
    const testState = {
        counterValue: 0
      };
    const store = createMockStore(testState);
    store.dispatch = jest.fn();
    const wrapper = mount( <Provider store={store}><CounterApp /></Provider> );
    it('renders without crashing', () => {
        const counterApp = renderer.create( <Provider store={store}><CounterApp /></Provider>).toJSON();
        expect(counterApp).toMatchSnapshot();
    });
    it('sets initial value of counter to passed prop', () => {
        expect(wrapper.find(Counter).props().counterValue).toEqual(testState.counterValue);
    });
    it('calls dispatch with "INCREMENT" action on clicking "ADD" button', () => {
        wrapper.find('.Add').simulate('click');
        expect(store.dispatch).toHaveBeenCalledWith(incrementActionGenerator());
    });
    it('calls dispatch with "DECREMENT" action on clicking "SUBTRACT" button', () => {
        wrapper.find('.Subtract').simulate('click');
        expect(store.dispatch).toHaveBeenCalledWith(decrementActionGenerator());
    });

})