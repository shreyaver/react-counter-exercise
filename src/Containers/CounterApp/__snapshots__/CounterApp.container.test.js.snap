// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp renders without crashing 1`] = `
<div
  className="Counter"
>
  <p
    className="Counter-display"
  >
    Counter: 
    0
  </p>
  <button
    className="Add"
    onClick={[Function]}
    style={
      Object {
        "color": "#fff",
      }
    }
    type="button"
  >
    Add
  </button>
  <button
    className="Subtract"
    onClick={[Function]}
    style={
      Object {
        "color": "#fff",
      }
    }
    type="button"
  >
    Subtract
  </button>
</div>
`;
